### PR TITLE
[EXP][CMDBUF] Implement Fill commands for L0 adapter

### PR DESCRIPTION
**Pull request by Bensuo**
Note: This PR also includes changes from the `main` branch (commit 40a6846ec8c65f3ee3ee3fee97ea4c27859ef193) which add these commands to the UR specification, but are not present in the adapter branch yet.

- Adds implementation of command-buffer fill commands for the L0 adapter.
- Includes stubs for the CUDA and HIP adapters.

### CODE REVIEW

**Comment by Bensuo** [comment]
Linked Intel/LLVM PR: https://github.com/intel/llvm/pull/11472

**Review by jandres742 — COMMENTED** [review]
changes looking good from L0 side

File: scripts/core/EXP-COMMAND-BUFFER.rst
Comment (mfrancepillois) [reply scripts/core/EXP-COMMAND-BUFFER.rst]: Done :+1:

**Comment by mfrancepillois** [comment]
This PR was rebased after the main branch was merged with adapter branch.
We also added the support for CUDA backend.

**Review by EwanC — APPROVED** [review]

File: source/adapters/cuda/command_buffer.cpp
Comment (mmoadeli) [inline source/adapters/cuda/command_buffer.cpp:116]: N can be `const size_t`

**Comment by EwanC** [comment]
> changes looking good from L0 side

Could you hit approve on this @jandres742 if you are still happy

**Comment by EwanC** [comment]
@nrspruit could you review on behalf of @oneapi-src/unified-runtime-level-zero-write

**Pull request merged**
